### PR TITLE
add countme option to yum_repository

### PR DESCRIPTION
**Pull request by konstruktoid**
##### SUMMARY

This PR:
- add `countme` support to the `yum_repository` module  (https://dnf.readthedocs.io/en/latest/conf_ref.html#options-for-both-main-and-repo)

##### ISSUE TYPE

- Feature Pull Request

### CODE REVIEW

File: lib/ansible/modules/yum_repository.py [hunk 1]
```diff
@@ -50,6 +50,11 @@
       - Relative cost of accessing this repository. Useful for weighing one
         repo's packages as greater/less than any other.
     type: str
+  countme:
+    description:
+      - Whether a special flag should be added to a randomly chosen query each week.
+      - Default value is V(false).
+      type: bool
```
Comment (Akasurde) [inline lib/ansible/modules/yum_repository.py]:
```suggestion
    description:
      - Whether a special flag should be added to a randomly chosen query each week.
      - Default value is V(false).
    type: bool
```

Comment (Akasurde) [reply lib/ansible/modules/yum_repository.py]: Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

File: lib/ansible/modules/yum_repository.py [hunk 2]
```diff
@@ -50,6 +50,12 @@
       - Relative cost of accessing this repository. Useful for weighing one
         repo's packages as greater/less than any other.
     type: str
+  countme:
+    description:
+      - Whether a special flag should be added to a randomly chosen query each week.
+      - Default value is V(false).
```
Comment (s-hertel) [inline lib/ansible/modules/yum_repository.py]:
```suggestion
    default: ~
```

Comment (konstruktoid) [reply lib/ansible/modules/yum_repository.py]: What does `~` mean in this context? Other modules use `default: false`

Comment (Akasurde) [reply lib/ansible/modules/yum_repository.py]: Argument spec default is None, `~` denotes that. If you want `default=False` you may want to specify that in the argument_spec.

File: lib/ansible/modules/yum_repository.py [hunk 3]
```diff
@@ -50,6 +50,12 @@
       - Relative cost of accessing this repository. Useful for weighing one
         repo's packages as greater/less than any other.
     type: str
+  countme:
+    description:
+      - Whether a special flag should be added to a randomly chosen query each week.
+    default: ~
+    type: bool
+    version_added: '2.17'
```
Comment (Akasurde) [inline lib/ansible/modules/yum_repository.py]:
```suggestion
    version_added: '2.18'
```

**Review by mkrizek — COMMENTED** [review]
One thing I'll note is that adding this might be potentially (more) confusing. The `countme` option is only available in dnf. I believe the `yum_repository` module was initially designed for yum and while it works (in part?) for dnf there is already a note in the documentation about RHEL7 vs RHEL8 scenario, and that is one we know about. Since the yum module was removed it might be a good time to review `yum_repository` options, perhaps rename the module to `dnf_repository`, $other.

EDIT: Also, dnf5.

**Comment by konstruktoid** [comment]
@mkrizek, are you thinking about https://docs.ansible.com/ansible/latest/collections/ansible/builtin/yum_repository_module.html#parameter-async? RHEL7 is EOL in about 3 months and the Ansible version has been EOL for a while.

Personally I don't have an strong opinion about renaming the module (except that it might break alot of roles depending on it), but it makes sense since no (?) RHEL-distro is only using `yum` today.
On the other hand, `dnf` doesn't have a repository directory but `yum` still do.

https://www.man7.org/linux/man-pages/man5/dnf.conf.5.html#DESCRIPTION:
```console
DNF by default uses the global configuration file at  /etc/dnf/dnf.conf and all *.repo files found under /etc/yum.repos.d. The latter is typically used for repository configuration and takes precedence over global configuration.
```

On AlmaLinux:
```sh
~$ cat /etc/redhat-release 
AlmaLinux release 8.9 (Midnight Oncilla)
~$ grep count /etc/yum.repos.d/*
/etc/yum.repos.d/almalinux-ha.repo:countme=1
/etc/yum.repos.d/almalinux-plus.repo:countme=1
/etc/yum.repos.d/almalinux-powertools.repo:countme=1
/etc/yum.repos.d/almalinux.repo:countme=1
/etc/yum.repos.d/almalinux.repo:countme=1
/etc/yum.repos.d/almalinux.repo:countme=1
/etc/yum.repos.d/almalinux-resilientstorage.repo:countme=1
/etc/yum.repos.d/almalinux-saphana.repo:countme=1
/etc/yum.repos.d/almalinux-sap.repo:countme=1
~$ ls -l $(which yum)
lrwxrwxrwx. 1 root root 5 Oct 17 20:04 /usr/bin/yum -> dnf-3
```

**Comment by konstruktoid** [comment]
i failed at branching @webknjaz :/

File: lib/ansible/modules/yum_repository.py [hunk 4]
```diff
@@ -50,6 +50,12 @@
       - Relative cost of accessing this repository. Useful for weighing one
         repo's packages as greater/less than any other.
     type: str
+  countme:
+    description:
+      - Whether a special flag should be added to a randomly chosen query each week.
```
Comment (mkrizek) [inline lib/ansible/modules/yum_repository.py]: Should we be more specific? Copy&pasting from the man page:
```suggestion
      - Whether a special flag should be added to a randomly chosen metalink/mirrorlist query each week.
        This allows the repository owner to estimate the number of systems consuming it. 
```

Comment (konstruktoid) [reply lib/ansible/modules/yum_repository.py]: Absolutely, I'll update later

File: lib/ansible/modules/yum_repository.py [hunk 5]
```diff
@@ -50,6 +50,13 @@
       - Relative cost of accessing this repository. Useful for weighing one
         repo's packages as greater/less than any other.
     type: str
+  countme:
+    description:
+      - Whether a special flag should be added to a randomly chosen metalink/mirrorlist query each week.
+        This allows the repository owner to estimate the number of systems consuming it.
```
Comment (mkrizek) [inline lib/ansible/modules/yum_repository.py]:
```suggestion
        This allows the repository owner to estimate the number of systems consuming it.
```

**Review by mkrizek — APPROVED** [review]

**Pull request merged**